task/sql-tasks.js

```diff
@@ -280,30 +463,62 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+  let result = await db.query(`
+        SELECT DISTINCT
+            C.CompanyName,
+            P.ProductName,
+            OD.UnitPrice as "PricePerItem"
+        FROM Customers as C
+        JOIN Orders as O ON O.CustomerID = C.CustomerID        
+        JOIN OrderDetails as OD ON OD.OrderID = O.OrderID
+        JOIN Products as P ON OD.ProductID = P.ProductID
+        WHERE OD.UnitPrice = (SELECT
+                                MAX(ODNew.UnitPrice)
+                              FROM Customers as CNew
+                              JOIN Orders as ONew ON ONew.CustomerID = CNew.CustomerID 
+                              JOIN OrderDetails as ODNew ON ONew.OrderID = ODNew.OrderID 
+                              WHERE CNew.CompanyName = C.CompanyName)

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.